File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/RangeRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public class RangeRule extends Rule<Integer> {

    private int minValue;
    private int maxValue;
    private boolean includeBounds;

    public RangeRule(Integer minValue, Integer maxValue, Boolean includeBounds) {
        super(Integer.class);
        this.minValue = minValue;
        this.maxValue = maxValue;
        this.includeBounds = includeBounds;
    }

    @Override
    public boolean isValid(Integer value) {
        if (includeBounds) {
            return value != null && value >= minValue && value <= maxValue;
        } else {
            return value != null && value > minValue && value < maxValue;
        }
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/RangeRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

class RangeRule(
    private val minValue: Int,
    private val maxValue: Int,
    private val includeBounds: Boolean
) : Rule<Int>(Int::class.java) {

    override fun isValid(value: Int?): Boolean {
        return if (includeBounds) {
            value != null && value >= minValue && value <= maxValue
        } else {
            value != null && value > minValue && value < maxValue
        }
    }
}
